packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -1667,4 +1697,9 @@ export class HardhatNode extends EventEmitter {
       filterId,
     });
   }
+
+  private async _isTransactionMined(hash: Buffer): Promise<boolean> {
+    const txReceipt = await this.getTransactionReceipt(hash);
+    return txReceipt !== undefined;
```

Comment:
**I'm not sure about this at all**. Is this enough? Or should we also consider a defined receipt with a falsy block number?